### PR TITLE
IDVA6-532 invitations to be digitally authorised page

### PR DESCRIPTION
**Pull request by mmodzelewski-ch**
Link to Jira:
https://companieshouse.atlassian.net/browse/IDVA6-532

Changes:
- Invitations page created
- Problems with debugger fixed
- ESLint upgraded and reconfigured
- Various issues listed by Lint fixed
- Code smells listed by SonarQube fixed

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/routers/handlers/yourCompanies/companyInvitationsHandler.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+export class CompanyInvitationsHandler extends GenericHandler {
+    async execute (req: Request): Promise<ViewData> {
+        this.viewData = await this.getViewData(req);
+        return Promise.resolve(this.viewData);
+    }
+
+    private async getViewData (req: Request): Promise<ViewData> {
+        const userEmailAddress = getLoggedInUserEmail(req.session);
+        const userAssociations: Associations = await getUserAssociations(userEmailAddress);
```
Comment (kkonuganti-ch) [inline src/routers/handlers/yourCompanies/companyInvitationsHandler.ts:19]: here you are using user email id to get associations.
We dont use useremail. We would just pass in the users details when calling the endpoint, which would already have current user details in session. We dont have an endpoint which would take user email.

File: src/routers/handlers/yourCompanies/companyInvitationsHandler.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        const awaitingApproval = userAssociations.items?.filter(item => item.status === AssociationStatus.AWAITING_APPROVAL);
```
Comment (kkonuganti-ch) [inline src/routers/handlers/yourCompanies/companyInvitationsHandler.ts:20]: the new endpoint we have will take in query parameters for status

**Review by kkonuganti-ch — APPROVED** [review]

File: src/routers/handlers/yourCompanies/companyInvitationsHandler.ts (hunk 1 above)
Comment (mmodzelewski-ch) [reply src/routers/handlers/yourCompanies/companyInvitationsHandler.ts:19]: All the code that is directly involved in API calls will be redone once all the APIs are available via api-sdk-node. There is no point making any changes before then.

File: src/routers/handlers/yourCompanies/companyInvitationsHandler.ts (hunk 2 above)
Comment (mmodzelewski-ch) [reply src/routers/handlers/yourCompanies/companyInvitationsHandler.ts:20]: All the code that is directly involved in API calls will be redone once all the APIs are available via api-sdk-node. There is no point making any changes before then.

**Pull request merged**